internal/chat/domain/chat_service.go

```diff
@@ -173,7 +173,31 @@ func (s *ChatServiceImpl) CreateChat(ctx context.Context, chat Chat) (*Chat, err
 }
 
 func (s *ChatServiceImpl) UpdateChat(ctx context.Context, chat Chat) (*Chat, error) {
-	updatedChat, err := s.chatRepo.UpdateChat(ctx, chat)
+	user := domain.UserFromContext(ctx)
+
+	existingChat, err := s.GetChat(ctx, chat.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if existingChat == nil {
+		return nil, errors.NewNotFoundError(common.ChatDomain)
+	}
+
+	if existingChat.CreatedBy != user.ID {
+		return nil, errors.NewForbiddenError()
+	}
+
+	tx, err := s.baseRepo.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		_ = tx.Rollback()
+	}()
+
+	updatedChat, err := s.chatRepo.UpdateChat(ctx, chat, tx)
```

Comment:
```suggestion
	updatedChat, err := s.chatRepo.UpdateChat(ctx, chat, nil)
```
Actually, transaction will not help on this way.